common/tasks/parallel_processor.go

```diff
@@ -119,8 +127,64 @@ func (p *ParallelProcessor) Submit(task Task) {
 	}
 }
 
-func (p *ParallelProcessor) processTask() {
-	defer p.workerWG.Done()
+func (p *ParallelProcessor) workerMonitor() {
+	defer p.shutdownWG.Done()
+
+	timer := time.NewTimer(backoff.JitDuration(defaultMonitorTickerDuration, defaultMonitorTickerJitter))
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-p.shutdownChan:
+			p.stopWorkers(len(p.workerShutdownCh))
+			return
+		case <-timer.C:
```

Comment:
I think we can define some generic dynamic config change listener that each component can register themself as listener to dynamic config for specific key. When value changed, the dynamic config will automatically call those registered listener to deliver the change. That way, we only have one place doing the periodically polling.

Reply:
Synced offline. Created an issue for supporting listener on dynamic config.
But will land this PR first so that in next release the value can be updated without restarting the service.  as well.